driver/src/withdraw_driver.rs

```diff
@@ -53,32 +73,34 @@ pub fn run_withdraw_listener<D, C>(db: &D, contract: &C) -> Result<(), Box<dyn E
     let slot = find_first_unapplied_slot(withdraw_slot + 1, contract)?;
         println!("Highest unprocessed withdraw_slot is {:?}", slot);
         if can_process(slot, contract)? {
-            println!("Processing withdraw_slot is {:?}", slot);
+            println!("Processing withdraw_slot {:?}", slot);
             let state_root = contract.get_current_state_root()?;
             let contract_withdraw_hash = contract.withdraw_hash_for_slot(slot)?;
-            let mut balances = db.get_current_balances(&state_root)?;
+            let balances = db.get_current_balances(&state_root)?;
 
             let withdraws = db.get_withdraws_of_slot(slot.low_u32())?;
-            // Hash withdraws and compare with contract
-            let mut withdraw_hash: H256 = H256::zero();
-            for pat in &withdraws {
-                withdraw_hash = pat.iter_hash(&mut withdraw_hash)
+            let withdraw_hash = withdraws.iter().fold(H256::zero(), |acc, w| w.iter_hash(&acc));
+            if withdraw_hash != contract_withdraw_hash {
+                return Err(Box::new(DriverError::new(
```

Comment:
I thought that a panic is appropriate here,
If I remember correctly, in the guidelines, it was stated that corrupted data ( which we would have in this case) is a reason for panic. 
Imagine we know the error over here and we know that something is wrong with our data, but then some other part of the program keeps on calculating a wrong status update..